### PR TITLE
testing: Utils test implemented in core module

### LINKED ISSUE #380: Add unit test for the Util.kt file in the core module
**Describe the solution you'd like**
Add unit test for the Util.kt

**Additional context**
#233 issue number adds a similar point.

**Would you like to work on the issue?**
Yes.

### PR DESCRIPTION
**Pull request by Ana2k**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straighforward changes).**

Fixes #380 

**Description**
Added UtilTest.kt file as unit test for Util.kt 

**Alternative(s) considered**
N/A

**Type**
Choose one:  Testing 

**Screenshots (if applicable)**
N/A

**Checklist**
- [x] I have read and acknowledge the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md)
- [x] I have read [How to Contribute](https://github.com/google/android-fhir/blob/master/docs/contributing.md)
- [x] I have read the [Developer's guide](https://github.com/google/android-fhir/wiki/Developer's-Guide)
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate )
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally
- [x] I have built and run the reference app(s) to verify my change fixes the issue and/or does not break the reference app(s)

### CODE REVIEW

**Comment by Ana2k** [comment]
@jingtang10 @deepankarb i have implemented this testing file. Please review this pr.

File: core/src/test/java/com/google/android/fhir/UtilTest.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.fhir
+
+import android.os.Build
+import java.time.ZoneId
+import java.time.format.DateTimeFormatter
+import java.util.*
```
Comment (epicadk) [inline core/src/test/java/com/google/android/fhir/UtilTest.kt:21]: Spotless check is failing most likely because of this.

Comment (Ana2k) [reply core/src/test/java/com/google/android/fhir/UtilTest.kt]: Thanks so much for the review @epicadk ! I have updated the changes, hope the build runs now. :)

File: core/src/test/java/com/google/android/fhir/UtilTest.kt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2021 Google LLC
```
Comment (epicadk) [inline core/src/test/java/com/google/android/fhir/UtilTest.kt]: Why this change?

File: core/src/test/java/com/google/android/fhir/UtilTest.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+@Config(sdk = [Build.VERSION_CODES.P])
+class UtilTest {
+  @Test
+  fun checkTimeZoneFormat() {
+    val currentDate = Date()
+    val simpleDateFormat =
+      DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSXXX", Locale.getDefault())
+        .withZone(ZoneId.systemDefault())
+    val expectedResult = simpleDateFormat.format(currentDate.toInstant())
+    assertEquals(currentDate.toTimeZoneString(), expectedResult)
```
Comment (jingtang10) [inline core/src/test/java/com/google/android/fhir/UtilTest.kt:40]: We use `Truth`: https://truth.dev/

File: core/src/test/java/com/google/android/fhir/UtilTest.kt [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+@RunWith(RobolectricTestRunner::class)
+@Config(sdk = [Build.VERSION_CODES.P])
+class UtilTest {
+  @Test
+  fun checkTimeZoneFormat() {
```
Comment (jingtang10) [inline core/src/test/java/com/google/android/fhir/UtilTest.kt:34]:
```suggestion
  fun shouldFormatDate() {
```

File: core/src/test/java/com/google/android/fhir/UtilTest.kt [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    val currentDate = Date()
```
Comment (jingtang10) [inline core/src/test/java/com/google/android/fhir/UtilTest.kt:35]: I'd rather use a fixed value in test to avoid flakiness.

**Review by jingtang10 — CHANGES_REQUESTED** [review]
thanks!

**Comment by jingtang10** [comment]
Is this still being worked on?

**Comment by Ana2k** [comment]
> Is this still being worked on?

Hi @jingtang10 I apologise for the delay. Sadly I had COVID for quite a while. I have just fully recovered. 
Really excited to start contributing again.
Will get this done ASAP.

**Comment by jingtang10** [comment]
> > Is this still being worked on?
> 
> Hi @jingtang10 I apologise for the delay. Sadly I had COVID for quite a while. I have just fully recovered.
> Really excited to start contributing again.
> Will get this done ASAP.

Hi @Ana2k - sorry to hear that you had covid but really pleased to hear that you're fully recovered. Please take it easy! Kind regards and best wishes to you!

**Comment by Ana2k** [comment]
@jingtang10 theres some updates in the Util.kt file and its test file in the master branch
Proceeding with updating test and date tests to pre existing UtilTest.kt
Please let me know if further changes are required in that file.

**Comment by Ana2k** [comment]
> @jingtang10 theres some updates in the Util.kt file and its test file in the master branch.
> The tests wrt Time and date were absent in pre existing UtilTest.kt.
> So, I am proceeding with updating time and date tests to it.
> Please let me know if further changes are required in that file.

@jingtang10 I have updated the shouldFormatDate function in utilTest.kt Kindly please review. Thanks!

**Review by jingtang10 — CHANGES_REQUESTED** [review]

File: engine/src/test/java/com/google/android/fhir/UtilTest.kt [hunk 6]
```diff
@@ -31,6 +35,19 @@ import org.robolectric.annotation.Config
...
+  fun shouldFormatDate() {
+    val fixedDate = "Sat Jun 05 16:17:31 +0530 2021"
+    val formatter = DateTimeFormatter.ofPattern("EE MMM dd HH:mm:ss ZZ yyyy")
+    val parsedDate = ZonedDateTime.parse(fixedDate, formatter)
+
+    val simpleDateFormat =
+      DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ssXXXXX", Locale.getDefault())
+        .withZone(ZoneId.systemDefault())
+    val expectedResult = simpleDateFormat.format(parsedDate.toInstant())
+    assertThat(parsedDate.toString()).isEqualTo(expectedResult)
```
[the remark this answers is not in the document]
Comment (Ana2k) [reply engine/src/test/java/com/google/android/fhir/UtilTest.kt:48]: Yes these are the changes for the same. The tests passed on my end. I hope no issues on this code snippet?

[the remark this answers is not in the document]
Comment (jingtang10) [reply engine/src/test/java/com/google/android/fhir/UtilTest.kt:48]: I'm not see the `Date.toTimeZoneString` method being called here. How is it being tested?

**Review by jingtang10 — CHANGES_REQUESTED** [review]

**Comment by Tarun-Bhardwaj** [comment]
Hi @Ana2k , is there an ETA you can provide for completing the changes as per the review comments in order to close this PR?

**Comment by Ana2k** [comment]
> Hi @Ana2k , is there an ETA you can provide for completing the changes as per the review comments in order to close this PR?

Hi @Tarun-Bhardwaj I am have almost completed working on this, ETA is around 11th August

**Comment by jingtang10** [comment]
Closing this since it's not been open worked on.

**Pull request closed without merging**